### PR TITLE
perf: memoize MapTooltip internals

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Issue

The internals of the tooltips re-render when you move the mouse even if it's on the same zone.

## Description

Memoizes the internal components of the map tooltip so they are stable, also memoizes the gauges, co2 square and estimated badge to be stable even between zones. This helps especially on zones which are fully estimated or have no data at all.

### Double check

- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: web/src/components/ZoneGauges.tsx [hunk 1]
```diff
@@ -49,3 +50,10 @@ export default function ZoneGaugesWithCO2Square({
     </div>
   );
 }
+
+export default memo(
+  ZoneGaugesWithCO2Square,
+  (prevProps, nextProps) =>
+    JSON.stringify(prevProps.zoneData) === JSON.stringify(nextProps.zoneData) &&
+    prevProps.withTooltips === nextProps.withTooltips
+);
```
Comment (VIKTORVAV99) [inline web/src/components/ZoneGauges.tsx:59]: The question here is if this costs more than it's worth. I'd be happy to delete it if that is the case.

Comment (VIKTORVAV99) [reply web/src/components/ZoneGauges.tsx]: Decided to remove it and move the empty object creation outside the function so it's stable.

File: web/src/features/map/MapTooltip.tsx [hunk 2]
```diff
@@ -9,33 +9,28 @@ import ZoneGaugesWithCO2Square from 'components/ZoneGauges';
...
 import { StateZoneData } from 'types';
 import { round } from 'utils/helpers';
 import { selectedDatetimeStringAtom } from 'utils/state/atoms';
 
 import { hoveredZoneAtom, mapMovingAtom, mousePositionAtom } from './mapAtoms';
 
-export function TooltipInner({
+const emptyZoneData: StateZoneData = {
+  p: {},
+  c: {},
```
Comment (madsnedergaard) [inline web/src/features/map/MapTooltip.tsx:22]: Is ci/fr/rr not needed on the empty data any longer?

**Review by madsnedergaard — APPROVED** [review]
LGTM, one minor question

File: web/src/features/map/MapTooltip.tsx (hunk 2 above)
Comment (VIKTORVAV99) [reply web/src/features/map/MapTooltip.tsx:22]: They where always optional as the internal function uses optional chaining, I don't really know why I added them last time.

**Pull request merged**